Issue caused by vertical-align on root node with rem/rex type

anil-hebbar opened the pull request:
#### 5fd497f250fc14bcbb42fcd5d39e167dca6f0b70
<pre>
Issue caused by vertical-align on root node with rem/rex type
<a href="https://bugs.webkit.org/show_bug.cgi?id=294451">https://bugs.webkit.org/show_bug.cgi?id=294451</a>
<a href="https://rdar.apple.com/151618114">rdar://151618114</a>

Reviewed by NOBODY (OOPS!).

The issue is when doing vertical-align of type rem or rex, it is assumed that there
is a valid root style. This is not so in case of root node and this was causing
issues (root style is NULL ofr root element).
The fix is in case the rootStyle is NULL, mark the property as invalid

* LayoutTests/fast/css/vertical-align-rem-on-root-expected.txt: Added.
* LayoutTests/fast/css/vertical-align-rem-on-root.html: Added.
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::CSSPrimitiveValue::convertToLength const):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5fd497f250fc14bcbb42fcd5d39e167dca6f0b70

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108554 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28215 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18639 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113763 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58953 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110517 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28904 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36769 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82449 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111502 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22937 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97778 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62885 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22354 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15918 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58478 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92307 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15969 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116884 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35608 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26251 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91473 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35981 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94050 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91274 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36167 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13933 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31371 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35510 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41045 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35222 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38571 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36900 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/css3/vertical-alaign-rem-on-root.html:
```diff
@@ -0,0 +1,27 @@
+            (async () => {
+                globalThis.testRunner?.waitUntilDone();
+                if (window.testRunner)
+                  testRunner.dumpAsText();
```
```suggestion
                testRunner?.dumpAsText();
```

nt1m commented on LayoutTests/css3/vertical-alaign-rem-on-root.html:
```diff
@@ -0,0 +1,27 @@
+                history.back()
+                await navigator.locks.query()
+                await navigator.storage.estimate()
+                document.documentElement.computedStyleMap().has('vertical-align')
```
we usually have semi colons at the end of lines

nt1m commented on LayoutTests/css3/vertical-alaign-rem-on-root.html:
fast/css would be a better directory. Also there's a typo (extra a in vertical-alaign)

anil-hebbar committed:
Issue caused by vertical-align on root node with rem/rex type
https://bugs.webkit.org/show_bug.cgi?id=294451
rdar://151618114

Reviewed by NOBODY (OOPS!).

The issue is when doing vertical-align of type rem or rex, it is assumed that there
is a valid root style. This is not so in case of root node and this was causing
issues (root style is NULL ofr root element).
The fix is in case the rootStyle is NULL, mark the property as invalid

* LayoutTests/fast/css/vertical-align-rem-on-root-expected.txt: Added.
* LayoutTests/fast/css/vertical-align-rem-on-root.html: Added.
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::CSSPrimitiveValue::convertToLength const):

weinig commented on LayoutTests/platform/mac-wk1/TestExpectations:
```diff
@@ -53,7 +53,7 @@ imported/w3c/web-platform-tests/webcodecs [ Skip ]
 # Features not supported in WebKit1
 #//////////////////////////////////////////////////////////////////////////////////////////
-
+fast/css/vertical-align-rem-on-root.html [ Skip ]
```
This should not be necessary.

anil-hebbar replied:
Webkit 1 does not support navigator.locks.query() and hence the test fails
